geoips/pydantic/sectors.py

```diff
@@ -0,0 +1,601 @@
+"""Pydantic models used to validate GeoIPS sector plugins."""
+
+from datetime import datetime
+from enum import Enum
+from typing import Literal, Tuple, Union
+from typing_extensions import Annotated
+
+from pydantic import (
+    Field,
+    ConfigDict,
+    field_validator,
+)
+from pydantic.functional_validators import AfterValidator
+
+from geoips.pydantic.bases import FrozenModel, PermissiveFrozenModel, PluginModel
+
+
+class EarthConstants(float, Enum):
+    """A simple class containing constant values relative to the geometry of Earth."""
+
+    SEMI_MAJOR_AXIS = 6371228.0  # Semimajor axis of the earth in meters.
+
+
+class XYCoordinate(FrozenModel):
+    """A coordinate in projection units."""
+
+    x: float = Field(None, description="The x coordinate in projection units.")
+    y: float = Field(None, description="The y coordinate in projection units.")
+
+
+def lat_lon_coordinate(arg: tuple[float, float]) -> tuple[float, float]:
+    """Validate a latitude and longitude coordinate."""
+    if arg[0] < -90 or arg[0] > 90:
+        raise ValueError("Latitude must be between -90 and 90")
+    if arg[1] < -180 or arg[1] > 180:
+        raise ValueError("Longitude must be between -180 and 180")
+    return arg
+
+
+LatLonCoordinate = Annotated[Tuple[float, float], AfterValidator(lat_lon_coordinate)]
+
+
+class SectorProjection(PermissiveFrozenModel):
+    """Projection information for a sector.
+
+    This is a dictionary that provides Proj projection information for the sector. For
+    more information on what parameters can be supplied, see the Proj documentation.
+
+    Validation has only been implemented for some of the most common options. If you
+    need validation for a parameter that is not currently implemented, please open an
+    issue and, if possible, a pull request on GitHub.
+    """
+
+    proj: str = Field(..., description="Proj projection alias.")
+    a: EarthConstants = Field(
```

Comment:
It's there some way to just validate using some pyresample or proj tools,  rather than explicitly implementing validation for specific pieces of projection information?

Reply:
Yes, we are considering how to handle this in the future. It looks like we can inspect the individual projections from the `proj` package and compare against its arguments to add the correct fields to the validator.

Reply:
Yes, definitely! This does not need to be handled here. I'm writing up an issue on this now.